src/widgets/widget_list.rs

```diff
@@ -0,0 +1,258 @@
+        WidgetListItem {
+            content,
+            style: Style::default(),
+        }
+    }
+
+    pub fn style(mut self, style: Style) -> WidgetListItem<E> {
+        self.style = style;
+        self
+    }
+
+    pub fn size_hint(&self, area: &Rect) -> Size {
+        self.content.size_hint(area)
+    }
+
+    /*pub fn width(&self) -> usize {
+        self.content.width()
+    }
+    */
+}
+
+/// A widget to display several items among which one can be selected (optional)
+///
+/// # Examples
+///
+/// ```
+/// # use ratatui::widgets::{Block, Borders, List, ListItem};
+/// # use ratatui::style::{Style, Color, Modifier};
+/// let items = [ListItem::new("Item 1"), ListItem::new("Item 2"), ListItem::new("Item 3")];
+/// List::new(items)
+///     .block(Block::default().title("List").borders(Borders::ALL))
+///     .style(Style::default().fg(Color::White))
+///     .highlight_style(Style::default().add_modifier(Modifier::ITALIC))
+///     .highlight_symbol(">>");
+/// ```
+#[derive(Debug, Clone)]
+pub struct WidgetList<'a, E: Widget + SizeHint + Clone> {
+    block: Option<Block<'a>>,
+    items: Vec<WidgetListItem<E>>,
+    /// Style used as a base style for the widget
+    style: Style,
+    start_corner: Corner,
+    /// Style used to render selected item
+    highlight_style: Style,
+    /// Symbol in front of the selected item (Shift all items to the right)
+    highlight_symbol: Option<&'a str>,
+    /// Whether to repeat the highlight symbol for each line of the selected item
+    repeat_highlight_symbol: bool,
+}
+
+impl<'a, E> WidgetList<'a, E>
+where
+    E: Widget + SizeHint + Clone,
+{
+    pub fn new<T>(items: T) -> WidgetList<'a, E>
+    where
+        T: Into<Vec<WidgetListItem<E>>>,
+        E: Widget + SizeHint + Clone,
+    {
+        WidgetList {
+            block: None,
+            style: Style::default(),
+            items: items.into(),
+            start_corner: Corner::TopLeft,
+            highlight_style: Style::default(),
+            highlight_symbol: None,
+            repeat_highlight_symbol: false,
+        }
+    }
+
+    pub fn block(mut self, block: Block<'a>) -> WidgetList<'a, E> {
+        self.block = Some(block);
+        self
+    }
+
+    pub fn style(mut self, style: Style) -> WidgetList<'a, E> {
+        self.style = style;
+        self
+    }
+
+    pub fn highlight_symbol(mut self, highlight_symbol: &'a str) -> WidgetList<'a, E> {
+        self.highlight_symbol = Some(highlight_symbol);
+        self
+    }
+
+    pub fn highlight_style(mut self, style: Style) -> WidgetList<'a, E> {
+        self.highlight_style = style;
+        self
+    }
+
+    pub fn repeat_highlight_symbol(mut self, repeat: bool) -> WidgetList<'a, E> {
+        self.repeat_highlight_symbol = repeat;
+        self
+    }
+
+    pub fn start_corner(mut self, corner: Corner) -> WidgetList<'a, E> {
+        self.start_corner = corner;
+        self
+    }
+
+    fn get_items_bounds(
```

Comment:
true, I will take a look at your PR and try to apply your feature

Reply:
@joshka I copied the your padding feature. the truncation with the paragraph is already there. I would keep the truncation to the used widget.

check please the following video and notice the **first item** while scrolling down. (it get truncated, but at the end). what do you think about the behavior? 

[![asciicast](https://asciinema.org/a/QxI1QWlWXpgUeNgr5QFZumxuZ.svg)](https://asciinema.org/a/QxI1QWlWXpgUeNgr5QFZumxuZ)

Reply:
I think it's probable that when item 9 is showing the video is showing just the first 3 lines of item 3 and chopping off the last line. It's difficult to check that as the body of the widget lines is all the same - can you add a line number / line count to make it obvious? (e.g. "this is a 3 line widget list item\nline 1: lorem ipsum...\nline2...")

In the changes for making this work for the normal list, I chopped off the last line of the last item (and ensured that the selected item would have not been chopped off). I'm not 100% happy with that approach, as I think the truncation should generally (at least as a default) be dependent on the scroll direction rather than just always truncating the last item. My PR linked above is more about trying to get some feedback on what works best.

Reply:
this video is better.
when `item3` is selected then `item0` shows 5/6 lines
`item4` => `item0` shows 2/6 lines

[![asciicast](https://asciinema.org/a/BLyUT3wOv5Bbw3ENrEqsK14ks.svg)](https://asciinema.org/a/BLyUT3wOv5Bbw3ENrEqsK14ks)



>  once the rendering happens it starts from the top rather than the middle of the widget being rendered.

yes. If the behaviour is bad, we can just do it like you did in #150

Reply:
Yeah, I think this is probably an ok initial state. I.e. we document how the truncation works and fix it up later to support configuring the truncation / partial display.